### PR TITLE
[NEXUS-8771] support @Transactional with automatic retry as necessary

### PR DESCRIPTION
**Pull request by mcculls**
https://issues.sonatype.org/browse/NEXUS-8771
http://bamboo.s/browse/NX3-OSSF452-2 :white_check_mark: 

See https://github.com/sonatype/nexus-oss/pull/1485 where it's applied to various storage layers.

### CODE REVIEW

**Comment by jdillon** [comment]
looking good, only comment I have so far is we should add some guarded trace logging to help when shit hits the fan and we need to understand what its doing in the field.

File: components/nexus-transaction/src/main/java/org/sonatype/nexus/transaction/Operations.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import java.lang.reflect.Method;
+
+import com.google.common.base.Throwables;
+import org.aopalliance.intercept.Joinpoint;
+
+/**
+ * Utility class that lets you wrap arbitrary sections of code as &#064;{@link Transactional}.
+ *
+ * <pre>
+ * UnitOfWork.transactional(new Operation&lt;T, SomeException&gt;()
```
Comment (kellyrob99) [inline components/nexus-transaction/src/main/java/org/sonatype/nexus/transaction/Operations.java]: UnitOfWork here is C&P from previous version?

Comment (mcculls) [inline components/nexus-transaction/src/main/java/org/sonatype/nexus/transaction/Operations.java]: yep :)

**Comment by kellyrob99** [comment]
+1 and especially appreciate the thorough unit tests

**Comment by mcculls** [comment]
Merged to master as f28482d

**Pull request closed without merging**